sympy/ntheory/residue_ntheory.py

```diff
@@ -1486,7 +1486,22 @@ def _val_poly(root, coefficients, p):
     return f_val % p
 
 
-def polynomial_congruence(coefficients, m):
+def _valid_expr(expr):
+    """This function is used by `polynomial_congruence`.
+    If `expr` is a univariate polynomial the it returns
+    its coefficients, otherwise it raises Valuerror
+    """
+
+    from sympy import Poly
+    if not expr.is_polynomial():
+        raise ValueError("The expression should be a polynomial")
+    polynomial = Poly(expr)
+    if not  polynomial.is_univariate:
+        raise ValueError("The expression should be univariate")
+    return polynomial.all_coeffs()
```

Comment:
It is expected that the coefficients be integers.

Reply:
Oh yes, there should be a check for that. I will add a check for it.

Reply:
Done.